pyvista/plotting/plotting.py

```diff
@@ -3857,18 +3857,22 @@ def _save_image(image, filename, return_img):
         if not image.size:
             raise ValueError('Empty image. Have you run plot() first?')
         # write screenshot to file if requested
-        if isinstance(filename, (str, pathlib.Path)):
+        if isinstance(filename, (str, pathlib.Path, io.BytesIO)):
             from PIL import Image
-            filename = pathlib.Path(filename)
-            if isinstance(pyvista.FIGURE_PATH, str) and not filename.is_absolute():
-                filename = pathlib.Path(os.path.join(pyvista.FIGURE_PATH, filename))
-            if not filename.suffix:
-                filename = filename.with_suffix('.png')
-            elif filename.suffix not in SUPPORTED_FORMATS:
-                raise ValueError(f'Unsupported extension {filename.suffix}\n' +
-                                 f'Must be one of the following: {SUPPORTED_FORMATS}')
-            image_path = os.path.abspath(os.path.expanduser(str(filename)))
-            Image.fromarray(image).save(image_path)
+
+            if isinstance(filename, (str, pathlib.Path)):
+                filename = pathlib.Path(filename)
+                if isinstance(pyvista.FIGURE_PATH, str) and not filename.is_absolute():
+                    filename = pathlib.Path(os.path.join(pyvista.FIGURE_PATH, filename))
+                if not filename.suffix:
+                    filename = filename.with_suffix('.png')
+                elif filename.suffix not in SUPPORTED_FORMATS:
+                    raise ValueError(f'Unsupported extension {filename.suffix}\n' +
+                                     f'Must be one of the following: {SUPPORTED_FORMATS}')
+                filename = os.path.abspath(os.path.expanduser(str(filename)))
+                Image.fromarray(image).save(filename)
+            else:
+                Image.fromarray(image).save(filename, format="PNG")
```

Comment:
Any chance we're missing out on other kinds of buffers here? Would it be a bad idea to just try to save as a PNG into whatever was passed?

If I understand correctly the implementation of saving isn't changing, exactly because it probably only assumes a buffer interface.

Reply:
Yeah, this was in the back of my mind while implementing this. I'm thinking we may just want to have a dedicated `screenshot_buffer()` method where we type check this and the user can specify the format so we don't force PNG on them.

Reply:
I don't have time to abstract this/cover other formats or buffer types. This is a good as I'm able to get on this at this time